core/src/main/java/com/google/zxing/aztec/encoder/State.java

```
@@ -70,6 +72,26 @@ int getBitCount() {
     return bitCount;
   }
 
+  State appendFLGn(int eci) {
+    // FIXME: validate eci in [0,999999]
+    State result = shiftAndAppend(HighLevelEncoder.MODE_PUNCT, 0); // 0: FLG(n)
+    Token token = result.token;
+    int bitsAdded = 3;
+    if (eci < 0) {
+      token = token.add(0, 3); // 0: FNC1
+    } else if (eci > 999999) {
+      // FIXME: throw error
```

You can throw an error here like IAE, why not?